src/main/java/com/linkedin/replica/wall/commands/impl/DeletePostCommand.java

```diff
@@ -26,34 +27,34 @@ public Object execute() throws ParseException {
+        validateArgs(new String[]{"postId"});
         // call dbHandler to get error or success message from dbHandler
         DateFormat format = new SimpleDateFormat("EEE MMM dd yyyy hh:mm a", Locale.ENGLISH);
 
         Post post;
         Gson googleJson = new Gson();
         String postId = args.get("postId").toString();
-        String authorId = args.get("authorId").toString();
-        String type = args.get("type").toString();
-        String companyId = args.get("companyId").toString();
-        String privacy = args.get("privacy").toString();
-        String text = args.get("text").toString();
-        Date timestamp = format.parse(args.get("timestamp").toString());
-        ArrayList<String> hashtags = googleJson.fromJson((JsonArray) args.get("hashtags"), ArrayList.class);
-        ArrayList<String> mentions = googleJson.fromJson((JsonArray) args.get("mentions"), ArrayList.class);
-        int likesCount = (int) args.get("likesCount");
-        ArrayList<String> images = googleJson.fromJson((JsonArray) args.get("images"), ArrayList.class);
-        ArrayList<String> videos = googleJson.fromJson((JsonArray) args.get("videos"), ArrayList.class);
-        ArrayList<String> urls = googleJson.fromJson((JsonArray) args.get("urls"), ArrayList.class);
-        int commentsCount = (int) args.get("commentsCount");
-        boolean isCompanyPost = (boolean) args.get("isCompanyPost");
-        boolean isPrior = (boolean) args.get("isPrior");
+//        String authorId = args.get("authorId").toString();
+//        String type = args.get("type").toString();
+//        String companyId = args.get("companyId").toString();
+//        String privacy = args.get("privacy").toString();
+//        String text = args.get("text").toString();
+//        Date timestamp = format.parse(args.get("timestamp").toString());
+//        ArrayList<String> hashtags = googleJson.fromJson((JsonArray) args.get("hashtags"), ArrayList.class);
+//        ArrayList<String> mentions = googleJson.fromJson((JsonArray) args.get("mentions"), ArrayList.class);
+//        int likesCount = (int) args.get("likesCount");
+//        ArrayList<String> images = googleJson.fromJson((JsonArray) args.get("images"), ArrayList.class);
+//        ArrayList<String> videos = googleJson.fromJson((JsonArray) args.get("videos"), ArrayList.class);
+//        ArrayList<String> urls = googleJson.fromJson((JsonArray) args.get("urls"), ArrayList.class);
+//        int commentsCount = (int) args.get("commentsCount");
+//        boolean isCompanyPost = (boolean) args.get("isCompanyPost");
+//        boolean isPrior = (boolean) args.get("isPrior");
         post = dbHandler.getPost(postId);
-
         String response = dbHandler.deletePost(post);
+        postsCacheHandler.deletePost(postId);
```

Comment:
if the post does not exist, there is no need to call redis command

Reply:
I think response should be boolean, what do you think?

Reply:
delete post function in cache checks before deleting from cache if it doesn't exist in cache it returns